src/store/modules/Linking/sagas/handleUrlSaga.ts

```
@@ -36,59 +38,61 @@ export function* handleUrlSaga(action: ReturnType<typeof actionCreator>) {
     return;
   }
 
+  waitForSelector(isSplashHiddenSelector);
+
   switch (path.toLowerCase()) {
-    case 'users':
-    case 'pinged':
-      // TODO: temp profile disabling
-      false &&
-        navigate({
-          name: 'UserProfile',
-          params: {userId: query.id ?? ''},
-        });
-      break;
     case 'browser':
-      openLinkWithInAppBrowser({url: query.url ?? ''});
-      break;
-    case 'followus':
-    case 'refjoined':
-    case 'announcements':
-      openLinkWithInAppBrowser({url});
+      if (query.url) {
+        const browserUrl = decodeURIComponent(query.url);
```

He meant to check if this is valid

---

what is valid? url? how do we check it?

---

That decode func should throw an error or return undefined if it's invalid

---

@ice-ares it decodes a string. any string is valid to it
`> decodeURIComponent("asdasd")
'asdasd'`

---

decodeURIComponent does not return an error for any input. However, it can throw an exception if the input string is not a properly encoded URI component. In that case, the function will throw a URIError.

For example, if decodeURIComponent is called with an invalid encoded string, such as %xy, it will throw a URIError.

---

What are we doing here?
In case of error during `decodeURIComponent` browser will not be opened (since next line won't be executed) and the error will be logged to sentry.

---

Just new URL(url) to validate it

---

done